indexing-service/src/main/java/org/apache/druid/indexing/common/task/AbstractTask.java

```diff
@@ -125,6 +139,69 @@ protected AbstractTask(
     this(id, groupId, taskResource, dataSource, context, IngestionMode.NONE);
   }
 
+  @Nullable
+  public String setup(TaskToolbox toolbox) throws Exception
```

Comment:
Can we add tests for this and the `cleanup` method? That should help with the Travis branch coverage failures.

Reply:
yes sir!